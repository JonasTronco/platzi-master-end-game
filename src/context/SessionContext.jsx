import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
