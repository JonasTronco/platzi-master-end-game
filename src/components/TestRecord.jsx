/* eslint-disable no-underscore-dangle */
import PropTypes from 'prop-types';
import { useSession } from '../context/SessionContext';
import AddIcon from '../assets/icons/add.svg';
import DownloadIcon from '../assets/icons/download.svg';

function TestRecord({ test }) {
  const { session } = useSession();

  // eslint-disable-next-line consistent-return
  function displayOptions() {
    if (session && session.user.typeOfUser === 'Administrator') {
      return (
        <td className="table__body__row__cell__options">
          <AddIcon className="table__body__row__cell__icon" />
          <DownloadIcon className="table__body__row__cell__icon" />
        </td>
      );
    }

    if (session && session.user.typeOfUser === 'Doctor') {
      return (
        <td className="table__body__row__cell__options">
          <DownloadIcon className="table__body__row__cell__icon" />
        </td>
      );
    }

    if (session && session.user.typeOfUser === 'Bacteriologist') {
      return (
        <td className="table__body__row__cell__options">
          <AddIcon className="table__body__row__cell__icon" />
        </td>
      );
    }

    if (session && session.user.typeOfUser === 'Patient') {
      return (
        <td className="table__body__row__cell__options">
          <DownloadIcon className="table__body__row__cell__icon" />
        </td>
      );
    }
  }
  const TestRecordOptions = displayOptions();

  return (
    <tr className="table__body__row">
      <td className="table__body__row__cell__user">
        <p>{test._id}</p>
      </td>
      <td className="table__body__row__cell">
        <p>{test.name}</p>
      </td>
      <td className="table__body__row__cell">
        <p>{test.appointmentDate}</p>
      </td>
      {TestRecordOptions}
    </tr>
  );
}

TestRecord.propTypes = {
  test: PropTypes.object.isRequired,
};

export default TestRecord;
