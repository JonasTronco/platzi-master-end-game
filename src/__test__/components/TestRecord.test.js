import TestRecord from '../../components/TestRecord';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { SessionProvider } from '../../context/SessionContext';
describe('TestRecord Component', () => {
  const mock = {
    _id: '5ecda8f2e67da2531795ceb6',
    name: '1,25-Dihydroxyvitamin D, Serum',
    shortName: 'DHVD',
    isComplete: true,
    appointmentDate: '2020-05-27T23:40:34.096Z',
    createdAt: '2020-05-26T23:40:34.096Z',
    resultDate: '2020-05-26T23:41:15.057Z',
    resultId: '5ecda91b615c1553c4ef1bb8'
  };
  test('Should render TestRecord component', () => {
    const component = create(
      <SessionProvider>
        <TestRecord test={mock} />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('<TestRecord />', () => {
    const component = mount(
      <SessionProvider>
        <TestRecord test={mock} />
      </SessionProvider>
    );
    expect(component.length).toEqual(1);
  });
});
