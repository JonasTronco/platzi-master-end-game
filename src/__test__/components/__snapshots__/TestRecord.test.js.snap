// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestRecord Component Should render TestRecord component 1`] = `
<tr
  className="table__body__row"
>
  <td
    className="table__body__row__cell__user"
  >
    <p>
      5ecda8f2e67da2531795ceb6
    </p>
  </td>
  <td
    className="table__body__row__cell"
  >
    <p>
      1,25-Dihydroxyvitamin D, Serum
    </p>
  </td>
  <td
    className="table__body__row__cell"
  >
    <p>
      2020-05-27T23:40:34.096Z
    </p>
  </td>
</tr>
`;
